core/document/src/lib.rs

```diff
@@ -1,17 +1,21 @@
 pub mod operation;
 
+use std::collections::{hash_map::Keys, HashMap};
+
 pub use kurbo::{Circle, Point, Rect};
 pub use operation::Operation;
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum SvgElement {
+	Folder(Folder),
 	Circle(Circle),
 	Rect(Rect),
 }
 
 impl SvgElement {
 	pub fn render(&self) -> String {
 		match self {
+			Self::Folder(f) => f.elements.values().map(|e| e.render()).fold(String::with_capacity(f.elements.len() * 30), |s, e| s + &e),
 			Self::Circle(c) => {
```

Comment:
Throughout the whole PR, could you please change the use of single-letter variable names into more descriptive names? I still need to find the time to write a code style guide. But basically `s` -> `segment`, `f` -> `folder`, `c` -> `circle`/`color`, `d` -> `document`, etc. However `i` in a for-loop is fine and `e` in an error are fine because those are well known "common" and "standard" variable names in their own right.